joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -67,6 +67,22 @@ controller_interface::CallbackReturn JointTrajectoryController::on_init()
       "allow_integration_in_goal_trajectories", allow_integration_in_goal_trajectories_);
     state_publish_rate_ = auto_declare<double>("state_publish_rate", 50.0);
     action_monitor_rate_ = auto_declare<double>("action_monitor_rate", 20.0);
+
+    std::string interpolation_method = "";
+    interpolation_method = auto_declare<std::string>("interpolation_method", "splines");
+    if (interpolation_method == "none")
```

Comment:
this if-else thing is basically a `from_string(... str)` method that could live alongside the class enum, helping with maintainability